File: JsonHelper/app/src/main/java/work/example/chris/jsonhelper/MainActivity.java
package work.example.chris.jsonhelper;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;

import work.example.chris.jsonhelper.model.Record;
import work.example.chris.jsonhelper.model.RecordSets;
import work.example.chris.jsonhelper.utils.JsonHelper;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = MainActivity.class.getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    protected void onStart() {

        Log.d(TAG, "onStart: ");

        super.onStart();

        RecordSets dataLists = JsonHelper.getModelFromRaw(this);

        for(Record item : dataLists.getRecord()) {
            Log.d(TAG, "Date: " + item.getCdate());
            Log.d(TAG, "Name: " + item.getName());
            Log.d(TAG, "Type: " + item.getType());
            Log.d(TAG, "Price: " + item.getPrice());
        }
    }
}

File: DeferredDeepLink/app/src/main/java/com/example/deferreddeeplink/view/MainActivity.kt
package com.example.deferreddeeplink.view

import android.content.Intent
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.example.deferreddeeplink.R
import io.branch.referral.Branch
import io.branch.referral.Branch.BranchReferralInitListener
import io.branch.referral.BranchError
import org.json.JSONObject

class MainActivity : AppCompatActivity(), BranchReferralInitListener {

    private val tag: String = MainActivity::javaClass.name

    override fun onInitFinished(referringParams: JSONObject?, error: BranchError?) {
        Log.i(tag, "onInitFinished")
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onStart() {
        super.onStart()

        val branch = Branch.getInstance()

        // Branch init
        branch.initSession({ referringParams, error ->
            if (error == null) {
                Log.i("BRANCH SDK", referringParams.toString())

                when (referringParams.optString("~id", "")) {
                    "" -> startActivity(Intent(this, SlaveActivity::class.java))
                    "666902580601505944" -> startActivity(Intent(this, ChildActivity::class.java))
                    "666866886290084734" -> startActivity(Intent(this, AnotherActivity::class.java))
                }
            } else {
                Log.i("BRANCH SDK", error.message)
            }
        }, this.intent.data, this)
    }

    override fun onNewIntent(intent: Intent) {
        this.intent = intent

        this.intent.let {
            val myIntent = it

            val action = myIntent.action
            val data = myIntent.data

            Log.i("Action: ", action.toString())
            Log.i("Data: ", data.toString())
        }
    }
}